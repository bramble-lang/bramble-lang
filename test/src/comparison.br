fn my_main() -> i64 {
    let b0 :bool := simple_eq(5, 6);
    project::std::io::writeboolln(b0);

    let b1:bool := big_equal(5, 5, true);
    project::std::io::writeboolln(b1);
    
    let b2:bool := big_equal(5, 7, true);
    project::std::io::writeboolln(b2);
    
    let b3:bool := lesser(3, 5);
    project::std::io::writeboolln(b3);

    let b4:bool := greater(3, 5);
    project::std::io::writeboolln(b4);

    let b5: bool := not_equal(3, 5);
    project::std::io::writeboolln(b5);
    
    return 0;
}

fn simple_eq(a: i64, b: i64) -> bool {
    return a == b;
}

fn not_equal(a: i64, b: i64) -> bool {
    return a != b;
}

fn big_equal(a: i64, b: i64, c: bool) -> bool{
    return (a == b) == c;
}

fn lesser(a: i64, b: i64) -> bool {
    return a < b;
}

fn greater(a: i64, b: i64) -> bool {
    return a > b;
}