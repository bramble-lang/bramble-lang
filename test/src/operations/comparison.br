fn my_main() {
    test(2);
    test(5);
    test(15);
    test(110);
    test(255);
    test(256);
    test(65535);
    test(65536);
    test(12342143144316);
    test(-12342143144316);

    return;
}

fn test(a: i64) {
    if (a < 10) {
        root::std::io::writei64ln(a)
    } else {
        root::std::io::write("failed\n")
    };

    return;
}