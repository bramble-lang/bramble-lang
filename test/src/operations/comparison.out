2
5
failed
failed
failed
failed
failed
failed
failed
-12342143144316
